### PR TITLE
Couple of Fixes for image download and Features 

### LINKED ISSUE #12: Fix for Training Images downloaded for outside AOI
Currently images are being downloaded also from neighbors area of AOI , Since they are resulting the extra images for the models , we need to find the bug and download only needed images inside AOI

### LINKED ISSUE #13: Feature : Download images also on the basis of zoom level
Plan is to Download the Images from API taking input from API zoom level  parameters directly from the POST requests

### LINKED ISSUE #14: Feature : Archive option for the Dataset
Plan is to create archive option for dataset from the API

### LINKED ISSUE #15: Store source link of imagery in DB
Plan is to store the link of imagery also for the training dataset

### PR DESCRIPTION
**Pull request by kshitijrajsharma**
- Resolve #15 
- Resolve #14 
- Resolve #13 
- Resolve #12 

Example request body to build image 
```
{"dataset_id":1,"source":"https://tiles.openaerialmap.org/5ac4fc6f26964b0010033112/0/5ac4fc6f26964b0010033113/{z}/{x}/{y}","zoom_level":19}
```
Current workflow for updating the dataset source 
1. Initially source will be null when creating dataset 
2. While posting request for image download it will seek for source , if not supplied it will use maxar as default else it will update the source supplied with dataset source_imagery 

 - Zoom level again is optional if not supplied it will assume as 19 and not stored in db 
 - Changes are backward compatible

### CODE REVIEW

File: backend/core/views.py [hunk 1]
```diff
@@ -196,16 +196,20 @@ def image_download_api(request):
...
     """
     print(request.data)
 
     serializer = ImageDownloadSerializer(data=request.data)
     if serializer.is_valid(raise_exception=True):
-        dataset_id = request.data.get("dataset_id")
+        dataset_id = int(request.data.get("dataset_id"))
         # get source imagery url if supplied else use maxar
         source = request.data.get("source", "maxar")
+        zoom_level = int(request.data.get("zoom_level", 19))
```
Comment (omranlm) [inline backend/core/views.py]: so the zoom is a parameter now and default value is 19

Comment (kshitijrajsharma) [reply backend/core/views.py]: Yes true !

**Comment by omranlm** [comment]
@kshitijrajsharma we need to have multiple zooms passed to the API and the API download all zoom(s) tiles from the zoom list

**Comment by kshitijrajsharma** [comment]
> @kshitijrajsharma we need to have multiple zooms passed to the API and the API download all zoom(s) tiles from the zoom list

Done ! 

```
{"dataset_id":1,"source":"https://tiles.openaerialmap.org/5ac4fc6f26964b0010033112/0/5ac4fc6f26964b0010033113/{z}/{x}/{y}","zoom_level":[19,20]}
```

File: backend/core/serializers.py [hunk 2]
```diff
@@ -1,38 +1,49 @@
...
-    
+    zoom_level = serializers.ListField(required=False)
+
+    def validate(self, data):
+        """
+        Check supplied data
+        """
+        for i in data['zoom_level']:
+            if int(i) < 19 or int(i) > 21:
+                raise serializers.ValidationError("Zoom level Supported between 19-21")
```
Comment (omranlm) [inline backend/core/serializers.py:43]: I like this one

**Review by omranlm — APPROVED** [review]
Great work, thanks @kshitijrajsharma

**Pull request merged**